tensorflow_datasets/core/dataset_builder.py

```diff
@@ -484,13 +519,13 @@ def as_dataset(self,
 
   def _build_single_dataset(
       self,
-      split,
-      shuffle_files,
-      batch_size,
-      decoders,
-      read_config,
-      as_supervised,
-      in_memory):
+      split: splits_lib.SplitBase,
```

Comment:
`Optional`

Reply:
There is no `split=None` So I don't think we should make it Optional

Reply:
Changed. Please check